flow/envs/base_env.py

```diff
@@ -633,6 +764,59 @@ def reset(self):
         for _ in range(self.env_params.warmup_steps):
             observation, _, _, _ = self.step(rl_actions=None)
 
+        if self.sumo_params.render in ["gray", "dgray", "rgb", "drgb"]:
```

Comment:
This code is duplicated elsewhere, suggesting that it should be a method.

Reply:
Can you elaborate on this?

Reply:
same as the above comment

Reply:
Same above.

Reply:
Solved.